File: Leaf-Linux/reboot_datetime/reboot_datetime.c

#include<stdio.h>
#include<time.h>

// crontab:
// @reboot /logs/reboot_datetime
//
// timedatectl

char s [100];

int main()
{
    time_t t;

    FILE *out = fopen ( "/logs/reboot_datetime.log", "a" );

    time ( &t );

    fputs ( "+++++++++++++++++++++++++++++++++++++++++\n\r", out );

    fputs ( "Reboot datetime: ", out );

    sprintf ( s, "%s", ctime(&t) );

    fputs ( s, out );

    fputs ( "-----------------------------------------\n\r\n\r", out );

	fclose ( out );

    return 0;
}

File: Leaf-Linux/- SystemModifications/reboot_datetime/reboot_datetime.py
#linux: /usr/bin/python3 /logs/reset_datetime.py
#
#crontab: @reboot (sh /usr/bin/python3 /logs/reset_datetime.py)


from datetime import datetime

with open('reboot_datetime.log', 'a') as f:

	# datetime object containing current date and time
	now = datetime.now()
	 
	s =  ""
	s += "+++++++++++++++++++++++++++++++++++++++++++\n"
	s += "Reset DateTime = " + str(now) + "\n"
	s += "-------------------------------------------\n\n"

	f.write(s)
